### ISSUE TITLE
Name in header isn't updated

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#14234]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Verify the header of a user who is on OOS  until Jan 16th

## Expected Result:
Show OOO till16 Jan, she updated it to that yesterday (and it shows it in the notification)

## Actual Result:
Shows as Jan 9

## Workaround:
unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?

- [ ] MacOS / Desktop

**Version Number:** Version 1.2.52-4
**Reproducible in staging?:** Needs reproduction
**Reproducible in production?:** Needs reproduction
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** 
<img width="674" alt="image (1)" src="https://user-images.githubusercontent.com/43996225/211954090-ceca4e91-e920-426b-9468-61be071fc2c4.png">

**Expensify/Expensify Issue URL:**
**Issue reported by:** @mallenexpensify 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1673476578716469

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by yuwenmemon** [comment]
I take a look at this and see if I can reproduce

**Comment by mallenexpensify** [comment]
Thanks @yuwenmemon , I'll post if it happens again. @alex-mechler can you too plz?

**Comment by alex-mechler** [comment]
Not sure why I was tagged here, I haven't reported this 😅

**Comment by yuwenmemon** [comment]
Oh, I was so confused. Yeah, the personal details are not being updated in the client. Something's up with that push notification.

**Comment by yuwenmemon** [comment]
@dangrous is there a reason why we only send a Push Notification to the user who is updating the Display Name [here](https://github.com/Expensify/Web-Expensify/blob/512817a5393eecea5df1d030cc0a2051637ec0e6/lib/PersonalDetailsAPI.php#L148)?

**Comment by yuwenmemon** [comment]
Looking at the [design doc](https://docs.google.com/document/d/1i8YQApRXOC1vWl4LMBUq4vHmA-58rNtVwn8gWpvLao0/edit#heading=h.517y4tfp9i8j) for the original refactor of this command - _maybe_ we just didn't think of that?

**Comment by Beamanator** [comment]
Yeah I don't think that was ever in scope (updating everyone else when I change my display name) - there's lots of places we _could_ do something like that (ex: updating a person's pronouns, timezone, and here we're talking about display name)

**Comment by mallenexpensify** [comment]
Updating names/statuses, timezones and others seems like something we'd _need_, right?  Especially if/once we actually have status updates.  Possibly too if we show someone is on/offline.  

> Not sure why I was tagged here, I haven't reported this 😅

Sorry @alex-mechler , I got this mixed up with the 3x Chronos message GH

**Comment by yuwenmemon** [comment]
Okay so fixing this is pretty simple I think. We should simply send pushes to anyone on your `personalDetailsList` (i.e. the value of `Get?rvl=personalDetailsList`) whenever you update your details. The thought is: if they're on your personalDetailsList, then you're on theirs (since you share a policy, domain, chat, etc.).

Right?

**Comment by yuwenmemon** [comment]
However, we need accountIDs rather than emails as the current [query](https://github.com/Expensify/Auth/blob/bd168885f4e136c823095be9d0cd83709ab6b297/auth/lib/Account.cpp#L1949-L2063) provides us. So we'd probably need to write a new Auth command for this.

**Comment by Beamanator** [comment]
Aah interesting - @yuwenmemon if you move forward implementing this, we may want to revisit paying contributors who reported things like Pronouns & Timezone not updating automatically when a user updates them?

I think I remembered discussing somewhere the concept of 1 channel message instead of many individual account messages - I will link the convo if I find it

**Comment by yuwenmemon** [comment]
Well, it would just be when updating your personal details, which in most use cases is a rare event. I would think a domain room, where you may be on a chat with 1000 other people and conversing quite frequently for instance, would be a bigger concern for "overusing" Pusher.

I will bring this up in Slack though to see if there are any other considerations with something like this though.

> Aah interesting - @yuwenmemon if you move forward implementing this, we may want to revisit paying contributors who reported things like Pronouns & Timezone not updating automatically when a user updates them?


What have we been saying in the past, that it's not a bug?

**Comment by yuwenmemon** [comment]
Raised in Slack [here](https://expensify.slack.com/archives/C01GTK53T8Q/p1673899235958819).

**Comment by yuwenmemon** [comment]
Well, that got a lot of traction! 😅 

I found myself thinking about the solution for this again today. I don't think we'd actually need a new Auth command for this. Any NewDot user will already have their `personalDetails` Onyx prop filled with all the emails of who they need to update. So maybe we can actually just send all those emails along with your updated display name? It's kind of weird... but the `personalDetails` Onyx prop is essentially a cache of the result of `Get?rvl=personalDetailsList`.

Gonna ask for thoughts about this in [#engineering-chat](https://expensify.slack.com/archives/C03TQ48KC/p1674009404217759).
